test/test_helpers/cinstance.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module TestHelpers
+  module Cinstance
+    def create_pending_cinstance
+      FactoryBot.create(:cinstance, plan: FactoryBot.create(:application_plan, approval_required: true))
+    end
```

Comment:
You can use a trait instead here https://github.com/3scale/porta/blob/5331e7de089c85a755817b7a0d47ab75b07ba311/test/factories/cinstance.rb

```ruby
FactoryBot.define do
  factory(:contract) do
    association :user_account, :factory => :pending_buyer_account

    after(:stub) do |cinstance|
      cinstance.created_at = 2.months.ago
      cinstance.state = 'live'
      cinstance.user_key ||= SecureRandom.hex(16)
      cinstance.provider_public_key ||= SecureRandom.hex(32)

      cinstance.stubs(:keys).returns([])
      cinstance.stubs(:access_rules).returns([])
    end

    after(:build) do |contract|
      if contract.plan && contract.user_account && contract.user_account.provider_account.nil?
        contract.plan.provider_account.buyer_accounts << contract.user_account
      end
    end
    
    trait :as_pending do
      transient do
        plan { FactoryBot.create(:application_plan, approval_required: true) }
      end
    end
  end

```

Reply:
Cool 👍🏻